Build error: undefined: os.CopyFS

When attempting to build Vib using `make build`, it fails with an error:

```console
$ make build
mkdir -p build
sed "s|%INSTALLPREFIX%|/usr/|g" core/plugins.in > core/plugins.go
go build -a -o build/vib
# github.com/vanilla-os/vib/api
api/download.go:195:13: undefined: os.CopyFS
make: *** [Makefile:12: build] Error 1
```

### Steps to reproduce:

1. `sudo apt install -y libbtrfs-dev libdevmapper-dev pkg-config build-essential`
2. `go get ./…`
3. `go get github.com/ebitengine/purego`
4. `make build`

### Environment:
+ Operating System: Debian 12
+ Go Version: 1.22.12

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -18,7 +18,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: 1.22
+        go-version-file: go.mod
 
     - name: Install Build Dependencies
       run: |

--- a/.github/workflows/docs.yml
+++ b/.github/workflows/docs.yml
@@ -14,7 +14,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version: 1.21
+          go-version-file: go.mod
 
       - name: Install pallas
         run: |

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -19,7 +19,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: 1.22
+        go-version-file: go.mod
 
     - name: Install Build Dependencies
       run: |

--- a/.github/workflows/vendored-release.yml
+++ b/.github/workflows/vendored-release.yml
@@ -19,7 +19,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v5
       with:
-        go-version: 1.22
+        go-version-file: go.mod
 
     - name: Vendor dependencies
       run: |

--- a/api/go.mod
+++ b/api/go.mod
@@ -1,3 +1,3 @@
 module github.com/vanilla-os/vib/api
 
-go 1.21
+go 1.23.0

--- a/go.mod
+++ b/go.mod
@@ -1,8 +1,6 @@
 module github.com/vanilla-os/vib
 
-go 1.22.0
-
-toolchain go1.23rc2
+go 1.24.0
 
 require github.com/spf13/cobra v1.8.1
 

